### PR TITLE
#2317303 | Create 'Send email' action

### PR DESCRIPTION
**Pull request by M1r1k**

### CODE REVIEW

**Comment by M1r1k** [comment]
It's not ready yet. It misses unit tests.

File: src/Plugin/Action/SendEmail.php [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    $key = 'rules_action_mail_' . $this->getPluginId();
+
+    $message = drupal_mail('rules', $key, $send_to, $params['langcode'], $params, $from);
+
+    if ($message['result']) {
+      $this->logger->log(LogLevel::NOTICE, $this->t('Successfully sent email to %recipient', array('%recipient' => $send_to));
+    }
+  }
+
+}
```
Comment (fago) [inline src/Plugin/Action/SendEmail.php]: Missing new line. :)

File: src/Plugin/Action/SendEmail.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ */
+class SendEmail extends RulesActionBase {
+
+  /**
+   * @var LoggerInterface $logger
+   */
+  protected $logger;
+
+  /**
+   * Constructs a CreatePathAlias object.
```
Comment (fago) [inline src/Plugin/Action/SendEmail.php]: wrong comment.

File: src/Plugin/Action/SendEmail.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ *       required = FALSE,
+ *     ),
+ *   }
+ * )
+ *
+ * @todo: Define that message Context should be textarea comparing with textfield Subject
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: Add group information from Drupal 7.
+ */
+class SendEmail extends RulesActionBase {
```
Comment (fago) [inline src/Plugin/Action/SendEmail.php]: I think you are missing the container aware plugin interface here.

File: src/Plugin/Action/SendEmail.php [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+ *     ),
+ *     "from" = @ContextDefinition("email",
+ *       label = @Translation("From"),
+ *       description = @Translation("The mail's from address. Leave it empty to use the site-wide configured address."),
+ *       required = FALSE,
+ *     ),
+ *     "language" = @ContextDefinition("language",
+ *       label = @Translation("Language"),
+ *       description = @Translation("If specified, the language used for getting the mail message and subject."),
+ *       required = FALSE,
```
Comment (fago) [inline src/Plugin/Action/SendEmail.php:45]: You are missing the default value here, or better said the @todo to add it. This as well as todos for marking some stuff as translatable. -> Everything that's not ported yet should have a todo.

File: src/Plugin/Action/SendEmail.php (hunk 3 above)
Comment (M1r1k) [inline src/Plugin/Action/SendEmail.php]: ContainerFactoryPluginInterface or ContainerAwareInterface?

**Comment by klausi** [comment]
This is now being worked on in #97

**Pull request closed without merging**